internal/providers/info/info.go

```diff
@@ -42,3 +42,8 @@ func NewUser(name, home, uuid, shell, gecos string, groups []Group) User {
 
 	return u
 }
+
+// Claimer is an interface that defines a method to extract the claims from the ID token.
+type Claimer interface {
```

Comment:
I don't like this interface name, in this context "Claim" is a noun, the name "Claimer" makes it sound like it can claim something.

How about we just name the interface "IDToken"?

Reply:
Although I also don't think that the `Claimer` name is perfect, I think we should keep the conventions [defined by Go](https://go.dev/doc/effective_go):

> By convention, one-method interfaces are named by the method name plus an -er suffix or similar modification to construct an agent noun: Reader, Writer, Formatter, CloseNotifier etc.

Sometimes this ends up in some weird names (like `Stringer`) but it does indicate quite well the method that the type must implement in order to satisfy the interface.

Reply:
I know the convention, but here it results in a misleading name. `Claim` can be both a noun and a verb. In this context it's a noun, but naming the interface `Claimer` gives the impression that the method name is a verb, i.e. that it can claim something. Anyway, if you still disagree, let's keep it as is.